pkg/plugins/scms/git/main.go

```diff
@@ -56,10 +59,18 @@ func New(s Spec) (*Git, error) {
 		s.Branch = "main"
 	}
 
-	return &Git{
-		spec:         s,
-		remoteBranch: git.SanitizeBranchName(s.Branch),
-	}, nil
+	g := Git{
+		spec: s,
+	}
+
+	switch s.DisableWorkingBranch {
+	case true:
+		g.HeadBranch = git.SanitizeBranchName(s.Branch)
+	case false:
+		g.HeadBranch = git.SanitizeBranchName(fmt.Sprintf("updatecli_%s", pipelineID))
+	}
```

Comment:
This logic should not be there, as it introduces a lot of coupling. I expect the code, responsible to call the `New` method, to pass the branch name in the provied spec.